core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/http/RequestExecutionContext.java

```diff
@@ -127,6 +130,14 @@ public void attemptMetricCollector(MetricCollector metricCollector) {
         this.attemptMetricCollector = metricCollector;
     }
 
+    public Optional<ProgressUpdater> progressUpdater() {
```

Comment:
Instead of handling null in a lot of places, can we just return a no-op progress updater if progressUpdater is null?

Reply:
done